build-system/tasks/runtime-test/runtime-test-base.js

```diff
@@ -92,7 +92,15 @@ function updateBrowsers(config) {
       },
     })
     .set('edge', {browsers: ['Edge']})
-    .set('firefox', {browsers: ['Firefox']})
+    .set('firefox', {
+      browsers: ['FirefoxHeadless'],
+      customLaunchers: {
+        FirefoxHeadless: {
+          base: 'Firefox',
+          flags: argv.headless ? ['-headless'] : [],
```

Comment:
Having `--headless` will need to be untangled with the Chrome_no_extensions_headless browser. For example - 

`gulp unit --firefox --headless` uses browsers `[ 'FirefoxHeadless' ]`
`gulp unit --headless --firefox` uses browsers `[ 'Chrome_no_extensions_headless' ]`

Reply:
Ooh, good catch. Seems like we can't use the convenient map logic anymore, so I've changed the flag parsing to prioritize non-Chrome browser flags and then other Chrome-related flags. Lemme know if this works.